_posts/en/posts/2016-06-21-csv-softfork-instructions.md

```diff
@@ -15,13 +15,19 @@ excerpt: There is an ongoing soft fork of the Bitcoin consensus rules. While eve
 
 There is an ongoing soft fork of the Bitcoin consensus rules. While everything appears to be proceeding well, this article contains important information and checklists for miners and pool operators which must not be ignored.
 
-If there is any doubt, miners and pool operators are welcome to [contact us][1].
+If there is any doubt, miners and pool operators are welcome to [contact us][].
 
 TL;DR 
 
-1. Check all your nodes have been upgraded to Bitcoin Core 0.12.1 or compatible software. This must happen before block #419328.
+Miners must make the following changes by block #419328.  Any miners who fail to make these changes may generate invalid blocks, causing them lost revenue and possibly disrupting other users and miners on the network.
 
-2. If you manually hardcode the block version field, or use the nSequence or nLockTime fields in the coinbase transaction, you must follow the instructions carefully.
+1. Check all your nodes have been upgraded to Bitcoin Core 0.12.1 or compatible software.
+
+1. If possible, automatically set the block version from Bitcoin Core.  If not possible, set the block version to 0x20000000.  You may do this any time before block 419328.  ([Details](#hardcoded-version))
+
+1. If you generate coinbase transactions whose nSequence is not 0, ensure that the nVersion of the coinbase transaction is 1.  ([Details](#nsequence))
+
+1. If you generate coinbase transactions where the nLockTime is greater than 500000000, ensure that whatever value you set it to passes the [BIP113][] rules.  (The commonly-seen value of 0xffffffff always passes the BIP113 rules.) ([Details](#nlocktime))
```

Comment:
The stuff about version is not correct. The instruction is to unset the bit. The TLDR is now really too long...